src/js/binaryen.js-post.js

```diff
@@ -1212,27 +1218,323 @@
+    var value;
+    switch (type) {
+      case Module['i32']: value = Module['_BinaryenConstGetValueI32'](expr); break;
+      case Module['i64']: value = { 'low': Module['_BinaryenConstGetValueI64Low'](expr), 'high': Module['_BinaryenConstGetValueI64High'](expr) }; break;
+      case Module['f32']: value = Module['_BinaryenConstGetValueF32'](expr); break;
+      case Module['f64']: value =  Module['_BinaryenConstGetValueF64'](expr); break;
+      default: throw Error('unexpected type: ' + type);
+    }
+    return {
+      'id': Module['ConstId'],
+      'type': type,
+      'value': value
+    };
+  };
+
+  Module['getUnary'] = function(expr) {
+    return {
+      'id': Module['UnaryId'],
+      'type': Module['_BinaryenExpressionGetType'](expr),
+      'op': Module['_BinaryenUnaryGetOp'](expr),
+      'value': Module['_BinaryenUnaryGetValue'](expr)
+    };
+  };
+
+  Module['getBinary'] = function(expr) {
+    return {
+      'id': Module['BinaryId'],
+      'type': Module['_BinaryenExpressionGetType'](expr),
+      'op': Module['_BinaryenBinaryGetOp'](expr),
+      'left': Module['_BinaryenBinaryGetLeft'](expr),
+      'right':  Module['_BinaryenBinaryGetRight'](expr)
+    };
   };
 
-  Module['getConstValueI64'] = function(expr) {
+  Module['getSelect'] = function(expr) {
     return {
-      'low': Module['_BinaryenConstGetValueI64Low'](expr),
-      'high': Module['_BinaryenConstGetValueI64High'](expr)
+      'id': Module['SelectId'],
+      'type': Module['_BinaryenExpressionGetType'](expr),
+      'ifTrue': Module['_BinaryenSelectGetIfTrue'](expr),
+      'ifFalse': Module['_BinaryenSelectGetIfFalse'](expr),
+      'condition': Module['_BinaryenSelectGetCondition'](expr)
     };
   };
 
-  Module['getConstValueF32'] = function(expr) {
-    return Module['_BinaryenConstGetValueF32'](expr);
+  Module['getDrop'] = function(expr) {
+    return {
+      'id': Module['DropId'],
+      'type': Module['_BinaryenExpressionGetType'](expr),
+      'value': Module['_BinaryenDropGetValue'](expr)
+    };
   };
 
-  Module['getConstValueF64'] = function(expr) {
-    return Module['_BinaryenConstGetValueF64'](expr);
+  Module['getReturn'] = function(expr) {
+    return {
+      'id': Module['ReturnId'],
+      'type': Module['_BinaryenExpressionGetType'](expr),
+      'value': Module['_BinaryenReturnGetValue'](expr)
+    };
   };
 
-  Module['getFunctionBody'] = function(func) {
-    return Module['_BinaryenFunctionGetBody'](func);
+  Module['getHost'] = function(expr) {
+    return {
+      'id': Module['HostId'],
+      'type': Module['_BinaryenExpressionGetType'](expr),
+      'op': Module['_BinaryenHostGetOp'](expr),
+      'nameOperand': Pointer_stringify(Module['_BinaryenHostGetNameOperand'](expr)),
+      'operands': getAllNested(expr, '_BinaryenHostGetNumOperands', '_BinaryenHostGetOperand')
+    }
+  };
+
+  Module['getNop'] = function(expr) {
+    assert(Module['_BinaryenExpressionGetId'](expr) == Module['NopId']);
```

Comment:
The reason for asserts only in nop/unreachable is that all the others call type specific functions that assert in C, while nops and unreachables don't, so should be asserted in JS.